src/io/syncsocket.c

```diff
@@ -306,41 +293,57 @@ struct sockaddr * MVM_io_resolve_host_name(MVMThreadContext *tc, MVMString *host
         case SOCKET_FAMILY_INET6:
             hints.ai_family = AF_INET6;
             break;
-        case SOCKET_FAMILY_UNIX:
-            hints.ai_family = AF_UNIX;
-            break;
+        case SOCKET_FAMILY_UNIX: {
+            /* TODO: UNIX socket support exists in newer versions of Windows.
+             *       See if it's good enough for us to use. */
+#ifndef AF_UNIX
+            MVM_free(host_cstr);
+            MVM_exception_throw_adhoc(tc, "UNIX sockets are not supported by MoarVM on this platform");
+#else
+            static const size_t MAX_SUN_LEN = sizeof(((struct sockaddr_un *)NULL)->sun_path);
+            size_t              sun_len     = strnlen(host_cstr, MAX_SUN_LEN);
+
+            if (sun_len >= MAX_SUN_LEN) {
+                char *waste[] = { host_cstr, NULL };
+                MVM_exception_throw_adhoc_free(
+                    tc, waste, "Socket path '%s' is too long (max supported by this platform is %zu)",
```

Comment:
This can be a bit confusing. Your length check correctly takes the terminating NULL character into account (not counted by strnlen, but offset by using `>=` instead of `>` in the comparison), so it will show an error message citing the 108 character path as too long due to a max of 108 characters supported.

Reply:
Should it say something like `"Socket path '...' is too long (max length supported by this platform is 107)"` instead?

Reply:
Yes, since that is the actual number that we allow.